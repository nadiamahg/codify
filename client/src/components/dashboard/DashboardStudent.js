import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { logoutUser } from "../../actions/authActions";
import { Link } from "react-router-dom";
import { getStudentAssignments } from "../../api/assignmentApi"

class DashboardStudent extends Component {

  constructor(props) {
    super(props)
    this.state = {
      assignments: [],
    }
  };

  componentDidMount = async () => {
    this.setState({ isLoading: true })

    await getStudentAssignments(this.props.auth.user.username).then(assignments => {
      this.setState({
        assignments: assignments.data.data
      })
    })
  };

 viewAssignment = (assignment_name) => {
    var route = '/' + assignment_name + '/assignment';
    this.props.history.push(route);
  }

  onLogoutClick = e => {
    e.preventDefault();
    this.props.logoutUser();
  };

  render() {
    const { user } = this.props.auth;
    const { assignments } = this.state;
    return (
      <div style={{ height: "75vh" }} className="container valign-wrapper">
        <div className="row">
          <div className="col s12 center-align">
            <h4>
              <b>Hey there,</b> {user.first_name} {user.surname}
              <p className="flow-text grey-text text-darken-1">
                You are logged into a full-stack{" "}
                <span style={{ fontFamily: "monospace" }}>MERN</span> app üëè
              </p>
            </h4>
            <table className="striped">
              <thead>
                <tr>
                  <th>Assignment Name</th>
                  <th>Score</th>
                  <th>Due Date</th>
                  <th>Status</th>
                </tr>
              </thead>
              <tbody>
                {
                  assignments.map((assignment, i) => {
                    return (
                      <tr key={i} onClick={this.viewAssignment.bind(this, assignment.assignment_name)}>
                        <td>
                          {assignment.assignment_name}
                        </td>
                        <td>
                          {assignment.assignment_score}
                        </td>
                        <td>
                          {assignment.assignment_due_date}
                        </td>
                        <td>
                          {assignment.assignment_status}
                        </td>
                      </tr>
                    );
                  })
                } 
              </tbody>
            </table>
            
            <button
              style={{
                width: "150px",
                borderRadius: "3px",
                letterSpacing: "1.5px",
                marginTop: "1rem"
              }}
              onClick={this.onLogoutClick}
              className="btn btn-large waves-effect waves-light hoverable blue accent-3"
            >
            Log Out
            </button>
          </div>
        </div>
      </div>



    );
  }
}
DashboardStudent.propTypes = {
  logoutUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired
};
const mapStateToProps = state => ({
  auth: state.auth
});
export default connect(
  mapStateToProps, { logoutUser }
)(DashboardStudent);