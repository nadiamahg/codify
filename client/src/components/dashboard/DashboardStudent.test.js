import React from 'react';
import renderer from 'react-test-renderer';
import DashboardStudent from './DashboardStudent';
import { BrowserRouter} from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});
const initialState = {
  auth: {}
};


import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux'
const mockStore = configureMockStore();
const store = mockStore(initialState);


describe('Test suits for DashboardStudent', () => {
it('renders', () => {
	const wrapper = shallow(<DashboardStudent store={store}/>);
	expect(wrapper).toMatchSnapshot();

 });
});
