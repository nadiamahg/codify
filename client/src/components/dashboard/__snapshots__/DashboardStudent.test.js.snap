// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for DashboardStudent renders 1`] = `ShallowWrapper {}`;
